### PR TITLE
MRG: update HOWTO_RELEASE document

### PR DESCRIPTION
**Pull request by matthew-brett**
Rather haphazard update of release notes from experience of releasing 1.9.3.

### CODE REVIEW

File: doc/HOWTO_RELEASE.rst.txt [hunk 1]
```diff
@@ -113,7 +119,7 @@ Wine
 For building Windows binaries on OS X Wine can be used. In Wine the following
 needs to be installed:
 
-* Python 2.6-2.7 and 3.2
+* Python 2.6-2.7 and 3.3
```
Comment (charris) [inline doc/HOWTO_RELEASE.rst.txt]: Note that numpy-vendor currently produces binaries for 2.7, 3.3, 3.4. We should add 3.5 and Carl's work at some point.

File: doc/HOWTO_RELEASE.rst.txt [hunk 2]
```diff
@@ -167,22 +173,27 @@ What is released
 
 Binaries
 --------
-Windows binaries in "superpack" form for Python 2.6/2.7/3.2/3.3.
-A superpack contains three builds, for SSE2, SSE3 and no SSE.
 
-OS X binaries are made in dmg format, targeting only the Python from
-`python.org <http://python.org>`_
+Windows binary instalers in "superpack" form for Python 2.7/3.3/3.4.  A
```
Comment (charris) [inline doc/HOWTO_RELEASE.rst.txt]: installers.

File: doc/HOWTO_RELEASE.rst.txt [hunk 3]
```diff
@@ -303,14 +319,41 @@ Identify the commit hash of the release, e.g. 1b2e1d63ff.
...
+
+     # Return the git revision as a string
+
+And make sure the ``VERSION`` variable is set properly.
+
+Now you can make the release commit::
+
+    git commit -s -m "REL: Release." setup.py
     git tag <version>
     git push origin <version>
```
Comment (charris) [inline doc/HOWTO_RELEASE.rst.txt]: Might want to defer this last step until the release succeeds. I've had to delete the tag upstream more than once.

File: doc/HOWTO_RELEASE.rst.txt [hunk 4]
```diff
@@ -303,14 +319,41 @@ Identify the commit hash of the release, e.g. 1b2e1d63ff.
...
 
+The ``-s`` flag makes a PGP (usually GPG) signed tag.  Please do sign the
+release tags.
```
Comment (charris) [inline doc/HOWTO_RELEASE.rst.txt]: Might include a link that explains how to register a signature using gpg.

File: doc/HOWTO_RELEASE.rst.txt (hunk 3 above)
Comment (charris) [inline doc/HOWTO_RELEASE.rst.txt]: Should also be clear what origin refers to here. For me it is my numpy fork on github.

File: doc/HOWTO_RELEASE.rst.txt (hunk 4 above)
Comment (charris) [inline doc/HOWTO_RELEASE.rst.txt]: Or maybe just note that a search will find useful information on doing so.

**Comment by charris** [comment]
Nice update, just a few nitpicks.

**Comment by charris** [comment]
I usually announce on Linux Weekly News as well, might add that. Just need to email the editor.

**Comment by charris** [comment]
The scipy.org announcement needs to be a PR at https://github.com/scipy/scipy.org. The file that needs modification is `www/index.rst`. Search for `News`.

**Comment by charris** [comment]
There is also facebook, g+, and twitter. I wonder if `scipy.org`  shouldn't have a presence in those places.

**Comment by charris** [comment]
@matthew-brett Be good to update this.

**Comment by matthew-brett** [comment]
Sorry - will do - been caught up in workey stuff.
​

File: doc/HOWTO_RELEASE.rst.txt (hunk 3 above)
Comment (rgommers) [inline doc/HOWTO_RELEASE.rst.txt]: +1 I've always created a branch including the ``REL: <change ISRELEASED in setup.py to True`` commit and then built all the binaries before pushing the release tag to Github.

**Comment by rgommers** [comment]
This all looks good to me, thanks Matthew.

**Comment by matthew-brett** [comment]
Rebased, edited - look OK?

File: doc/HOWTO_RELEASE.rst.txt [hunk 5]
```diff
@@ -34,24 +36,39 @@ Release Scripts
...
+this `OSX wheel building summary
+<https://github.com/MacPython/wiki/wiki/Spinning-wheels>`_ for details.
 
 Windows
 -------
+
+32-bit Python 2.7, 3.3, 3.4 are the versions for which we provide binary
+installers.
+
 Windows XP, Vista and 7 are supported.
```
Comment (charris) [inline doc/HOWTO_RELEASE.rst.txt]: Might remove this blank line and merge above an below parts, similar to the OS X documentation.

File: doc/HOWTO_RELEASE.rst.txt (hunk 1 above)
Comment (charris) [inline doc/HOWTO_RELEASE.rst.txt]: Hmm. Numpy-vendor has 2,6, 2.7, 3.2, 3.3, and 3.4. Probably we only need 2.7, 3.3, and 3.4 these days.

Comment (rgommers) [inline doc/HOWTO_RELEASE.rst.txt]: Indeed, we haven't been building 2.6 and 3.2 binaries for quite a while.

Comment (rgommers) [inline doc/HOWTO_RELEASE.rst.txt]: and that would make it consistent with the text on superpack installers below.

File: doc/HOWTO_RELEASE.rst.txt [hunk 6]
```diff
@@ -327,19 +409,41 @@ For example::
...
+.. _push-tag-and-commit:
+
+Push the release tag and commit
+-------------------------------
+
+Finally, now you are confident this tag correctly defines the source code that
+you released you can push the tag and release commit up to github::
+
+    git push  # Push release commit
+    git push origin <version>  # Push tag named <version>
```
Comment (rgommers) [inline doc/HOWTO_RELEASE.rst.txt]: Can you change "origin" to "upstream"? That's how we use it in http://docs.scipy.org/doc/numpy-dev/dev/gitwash/development_workflow.html#development-workflow, and most people will have "origin" point to their personal fork on Github.

**Comment by rgommers** [comment]
Went through it again, LGTM except for the 2 minor comments I made.

**Comment by charris** [comment]
Thanks Matthew.

**Pull request merged**